src/web/backend/ldap/api.py

```diff
@@ -18,6 +20,85 @@ def handle_not_found(e):
     return jsonify({'message': str(e)})
 
 
+class UserListViewSet(EdapMixin, MethodView):
+
+    def get(self):
+        """ List users """
+        res = self.edap.get_users()
+        data = edap_users_schema.load(res).data
+        return jsonify(api_users_schema.dump(data).data)
+
+    def post(self):
+        """ Create user """
+        username = request.json.get('uid')
+        password = request.json.get('password')
+        name = request.json.get('name')
+        surname = request.json.get('surname')
+        if not all([username, password, name, surname]):
+            return jsonify({'message': 'username, password, name, surname fields are required'}), 400
+
+        user = api_user_schema.load(request.json).data
+        try:
+            res = user.create(password=password)
+        except ConstraintError as e:
+            return jsonify({'message': "Failed to create user. {}".format(e)}), 400
+
+        return jsonify(res)
+
+
+class UserRetrieveViewSet(EdapMixin,
+                          MethodView):
+    """ ViewSet for single user """
+    def get(self, username):
+        """ List users """
+        try:
+            user = edap_user_schema.load(self.edap.get_user(username)).data
+        except MultipleObjectsFound:
+            return jsonify({'message': 'More than 1 user found'}), 409
+        except ObjectDoesNotExist:
+            return jsonify({'message': 'User does not exist'}), 404
+        user_groups = self.edap.get_user_groups(username)
+        user = {
+            **api_user_schema.dump(user).data,
+            "groups": [group for group in user_groups],
+            "franchises": api_franchises_schema.dump(user.get_franchises()).data,
+            "divisions": api_divisions_schema.dump(user.get_divisions()).data,
+            "teams": api_teams_schema.dump(user.get_teams()).data
+        }
+        return jsonify(user)
+
+    def delete(self, username):
+        """ Delete user """
+        self.edap.delete_user(username)
```

Comment:
Ideally, the call should be more robust, s.a.

```
  result = dict(success=True)          
  try:
      self.edap.delete_user(username)
  except Exception as exc:
      result["success"] = False
      result["message"] = f"Failed to delete {username} from LDAP: {str(exc)}"
                                       
  return jsonify(result), 202
```

Reply:
But doing so, we catch all the exceptions, without logging them and also we can pass sensitive information from exception to user, is that ok ?

Reply:
You are right. This is just a pitch, details don't have to be part of the message. The exception is not caught at all in the current code, and this looks like a nice place to catch it. Of course, it can be passed, so the higher level catches it and it is logged, or the logging can be added right to the catch code.

Reply:
Right now nextcloud or edap api don't raise any exceptions, which we don't catch, so if any exceptions will happen, they must have been because of error in code, so it should be fixed asap. 
And for such exceptions, we could add sentry, which will log all of them.
Front-end from it's side will receive 500 response from server, and write default error message "Something wrong happened", without details. 
Will it work ?

Reply:
The `ldap` module raises an exception if you pass it a bad request. Prior to your fix, the server backtraced and there was no error message.

Reply:
@matejak in this commit [commit](https://github.com/EnterpriseyIntranet/teap/pull/26/commits/a193b5cd9ddcb2a4ca5c27b7a4f99af7eca0ad8e) with try/except, I catch it and pass to user, this is what we need, isn't it? 
We can also add `NO_SUCH_OBJECT` ldap exception to edap, and catch it here, but it needs to be added to edap first.

Reply:
Yes, I agree, we don't need more to handle this. Do you also agree with my observation, i.e. are we on one page?